File: source/livana-be/src/main/java/octguy/livanabe/dto/response/ExperienceListingResponse.java
package octguy.livanabe.dto.response;

import lombok.Builder;
import lombok.Data;
import octguy.livanabe.dto.dto.ImageOrderResponse;
import octguy.livanabe.dto.dto.ListingHostDto;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

@Data
@Builder
public class ExperienceListingResponse {

    private UUID listingId;

    private ListingHostDto host;

    private String title;

    private BigDecimal price;

    private String description;

    private int capacity;

    private String address;

    private Double latitude;

    private Double longitude;

    private ExperienceCategoryResponse experienceCategory;

    private List<ImageOrderResponse> images;
}

File: source/livana-fe/src/types/response/experienceListingResponse.ts
import type { ExperienceCategoryResponse } from "./experienceCategoryResponse";

export interface ImageOrderResponse {
  image: {
    url: string;
    publicId: string;
  };
  order: number;
}

export interface ListingHostDto {
  hostId: string;
  hostDisplayName: string;
  avatarUrl: string;
  phoneNumber: string;
}

export interface ExperienceListingResponse {
  listingId: string;
  host: ListingHostDto;
  title: string;
  price: number;
  description: string;
  capacity: number;
  address: string;
  latitude: number;
  longitude: number;
  experienceCategory: ExperienceCategoryResponse;
  images: ImageOrderResponse[];
}
